src/hotspot/share/opto/superword.cpp

```diff
@@ -2346,6 +2348,204 @@ void SuperWord::verify_packs() {
 }
 #endif
 
+class PacksetGraph {
+private:
+  // pid: packset graph node id.
+  GrowableArray<int> _pid; // Node.idx -> pid
+  GrowableArray<int> _incnt;
+  GrowableArray<GrowableArray<int>> _out;
+  int _max_pid = 0;
+
+  GrowableArray<Node*> &_block;
+  GrowableArray<Node_List*> &_packset;
+  DepGraph &_dg;
+public:
+  PacksetGraph(Arena* a,
+                GrowableArray<Node*> &block,
+                GrowableArray<Node_List*> &packset,
+                DepGraph &dg)
+  : _pid(a, 8, 0, /* default */ 0), _block(block), _packset(packset), _dg(dg) {
+  }
+  // Get pid, if there is a packset node that n belongs to. Else return 0.
+  int get_pid_or_zero(const Node* n) const {
+    if ((int)n->_idx >= _pid.length()) {
+      return 0;
+    } else {
+      return _pid.at(n->_idx);
+    }
+  }
+  int get_pid(const Node* n) {
+    int poz = get_pid_or_zero(n);
+    assert(poz != 0, "pid should not be zero");
+    return poz;
+  }
+  void set_pid(const Node* n, int pid) {
+    assert(n != nullptr && pid > 0, "sane inputs");
+    _pid.at_put_grow(n->_idx, pid);
```

Comment:
This could be huge waste of space. `_idx` could be very big number vs mach smaller `_max_pid`.
Can we use `_bb_idx` instead?